ElectronClient/app/gui/NotebookPropertiesDialog.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+
+const { _ } = require('lib/locale.js');
+const { themeStyle } = require('../theme.js');
+const DialogButtonRow = require('./DialogButtonRow.min');
+
+const Folder = require('lib/models/Folder.js');
+
+interface NotebookPropertiesDialogProps {
+	theme: number,
+	folderId: string,
+	onClose: Function,
+}
+
+export default function NotebookPropertiesDialog(props: NotebookPropertiesDialogProps) {
+	console.info('Render NotebookPropertiesDialog');
+
+	const [notebook, setNotebook] = useState<any>({});
+	const [icon, setIcon] = useState<string>('fa fa-book'); //
+	const [loaded, setLoaded] = useState<boolean>(false);
+
+	const theme = themeStyle(props.theme);
+
+	useEffect(() => {
+		async function fetchNotebook() {
+			if (!props.folderId) {
+				setNotebook({notebook: null});
+			} else {
+				const n = await Folder.load(props.folderId);
+				const nb = formNotebook(n);
+				setNotebook(nb);
+			}
+		}
+
+		fetchNotebook();
+	}, [props.folderId]);
+
+	const formNotebook = (targetNotebook:any) => {
```

Comment:
Not sure about the purpose of this function. You've already defined a `setNotebook` (should be called `setFormFolder`) so this seems redundant. I think you can do all the loading and setting state in the useEffect above.

Reply:
I figured that out in my current test branch. Just need to work out the kinks before pushing it.